packages/web/src/hooks/useErrorPageOnFailedStatus.ts

```diff
@@ -0,0 +1,23 @@
+import { useEffect } from 'react'
+
+import { Status } from '@audius/common'
+
+import { ERROR_PAGE } from 'utils/route'
+
+import { useNavigateToPage } from './useNavigateToPage'
+
+/** Automatically navigates to the `/error` route if the provided status is
+ * `Status.ERROR`. NOTE: If the status passed to this hook is derived from an
+ * `audius-query` data fetching hook, the page using this hook needs to support
+ * resetting of the fetch state on mount (typically through the `force` option
+ * of `useQuery`). Otherwise, the user will be immediately redirected back to the
+ * error page if they hit 'Try Again'.
+ */
+export const useErrorPageOnFailedStatus = (status: Status) => {
```

Comment:
do we want to have this send a log to amplitude/sentry or anything? or is that already handled somewhere internally in audius-query

Reply:
I should add a comment. It is already being logged to sentry by audius-query. I didn't want to double log it.
Any other method of showing the error page (i.e. throwing) _will_ cause it to be logged again.